superflore/generators/bitbake/gen_packages.py

```diff
@@ -49,22 +49,58 @@ def regenerate_pkg(
     component_name = yoctoRecipe.convert_to_oe_name(
         distro.release_packages[pkg].repository_name)
     recipe = yoctoRecipe.convert_to_oe_name(pkg)
-    # check for an existing recipe
-    prefix = '{0}/meta-ros{1}-{2}/generated-recipes/{3}/{4}'.format(
-        repo_dir,
+    # check for an existing recipe which was removed by clean_ros_recipe_dirs
+    prefix = 'meta-ros{0}-{1}/generated-recipes/*/{2}_*.bb'.format(
         yoctoRecipe._get_ros_version(distro.name),
         distro.name,
-        component_name,
-        recipe,
+        recipe
     )
-    existing = glob.glob('{}_*.bb'.format(prefix))
+    existing = overlay.repo.git.status('--porcelain', '--', prefix)
+    if existing:
+        # The git status --porcelain output will look like this:
+        # D  meta-ros2-eloquent/generated-recipes/variants/ros-base_0.8.3-1.bb
+        # we want just the path with filename
+        if len(existing.split()) > 2:
+            warn('More than 1 recipe was found in ""git status --porcelain '
+                 'meta-ros{0}-{1}/generated-recipes/*/{2}_*.bb": "{3}"'
+                 .format(
+                     yoctoRecipe._get_ros_version(distro.name),
+                     distro.name,
+                     recipe,
+                     existing))
```

Comment:
I think this should be considered an assertion failure and therefore, the program should abort.

Reply:
But that will require to remove the already duplicated pcl recipe manually before running superflore, otherwise it would fail and abort.

Reply:
Then please update the comment on lines 60-62 for the case when there is more than one line output. Is the first token guaranteed to always be `D`? If not, then line 71 will need correcting.

Reply:
`found in` -> `output by`